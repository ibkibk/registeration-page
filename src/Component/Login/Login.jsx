import React, { Component } from "react";
import firebase, { providers } from "../../firebase";

const initialState = {
  user: null,
  email: "",
  password: "",
  fireErrors: "",
  emailError: "",
  passwordError: "",
};

class SignInForm extends Component {
  constructor() {
    super();

    this.state = initialState;

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  // validate = () => {
  //   let emailError = "";
  //   let passwordError = "";

  //   if (!this.state.password.length) {
  //     passwordError = "Password required";
  //   }

  //   if (!this.state.email.includes("@")) {
  //     emailError = "invalid email";
  //   }

  //   if (emailError || passwordError) {
  //     this.setState({ emailError, passwordError });
  //     return false;
  //   }

  //   return true;
  // };

  handleChange(e) {
    let target = e.target;
    let value = target.type === "checkbox" ? target.checked : target.value;
    let name = target.name;

    this.setState({
      [name]: value,
    });
  }

  signIn = () => {
    firebase
      .auth()
      .signInWithPopup(providers.google)
      .then((result) => {
        this.setState({ user: result.user });
      })
      .catch((error) => {
        console.log(error);
      });
  };

  handleSubmit(e) {
    e.preventDefault();
    firebase
      .auth()
      .signInWithEmailAndPassword(this.state.email, this.state.password)
      .catch((error) => {
        console.log(error);
        this.setState({ fireErrors: error.message });
      });
    // const isValid = this.validate();
    // if (isValid) {
    //   console.log(this.state);
    //   this.setState(initialState);
    // } else {
    //   return false;
    // }
  }

  render() {
    let errorNotification = this.state.fireErrors ? (
      <div className="Error"> {this.state.fireErrors} </div>
    ) : null;
    return (
      <div className="FormCenter">
        {errorNotification}
        <form onSubmit={this.handleSubmit} className="FormFields">
          <div className="FormField">
            <label className="FormField__Label" htmlFor="email">
              E-Mail Address
            </label>
            <input
              type="email"
              id="email"
              className="FormField__Input"
              placeholder="Enter your email"
              name="email"
              value={this.state.email}
              onChange={this.handleChange}
            />
            <div style={{ fontSize: 12, color: "red" }}>
              {this.state.emailError}
            </div>
          </div>

          <div className="FormField">
            <label className="FormField__Label" htmlFor="password">
              Password
            </label>
            <input
              type="password"
              id="password"
              className="FormField__Input"
              placeholder="Enter your password"
              name="password"
              value={this.state.password}
              onChange={this.handleChange}
            />
            <div style={{ fontSize: 12, color: "red" }}>
              {this.state.passwordError}
            </div>
          </div>

          <div className="FormField">
            <button onClick={this.handleSubmit} className="FormField__Button ">
              Sign In
            </button>{" "}
            <button onClick={this.signIn} to="/" className="FormField__Button ">
              Sign In with google
            </button>
          </div>
        </form>
      </div>
    );
  }
}

export default SignInForm;
