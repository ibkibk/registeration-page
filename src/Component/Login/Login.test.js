import React from "react";
import { shallow } from "enzyme";
import Login from "./Login";

describe("<Login /> with other props", () => {
  const initialProps = {
    email: "acesmndr@gmail.com",
    password: "notapassword",
  };
  const container = shallow(<Login {...initialProps} />);

  it("should have proper props for email field", () => {
    expect(container.find('input[type="email"]').props()).toEqual({
      type: "email",
      id: "email",
      className: "FormField__Input",
      placeholder: "Enter your email",
      name: "email",
      value: "acesmndr@gmail.com",
      onChange: expect.any(Function),
    });
  });

  it("should have proper props for password field", () => {
    expect(container.find('input[type="password"]').props()).toEqual({
      type: "password",
      id: "password",
      className: "FormField__Input",
      placeholder: "Enter your password",
      name: "password",
      value: "notapassword",
      onChange: expect.any(Function),
    });
  });
});
