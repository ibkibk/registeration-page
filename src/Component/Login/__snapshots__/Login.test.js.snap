// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> with no props should match the snapshot 1`] = `"<div class=\\"FormCenter\\"><form class=\\"FormFields\\"><div class=\\"FormField\\"><label class=\\"FormField__Label\\" for=\\"email\\">E-Mail Address</label><input type=\\"email\\" id=\\"email\\" class=\\"FormField__Input\\" placeholder=\\"Enter your email\\" name=\\"email\\" value=\\"\\"/></div><div class=\\"FormField\\"><label class=\\"FormField__Label\\" for=\\"password\\">Password</label><input type=\\"password\\" id=\\"password\\" class=\\"FormField__Input\\" placeholder=\\"Enter your password\\" name=\\"password\\" value=\\"\\"/></div><div class=\\"FormField\\"><button class=\\"FormField__Button \\">Sign In</button> <button to=\\"/\\" class=\\"FormField__Button \\">Sign In with google</button></div></form></div>"`;
